lib/runners/processor/detekt.rb

```diff
@@ -0,0 +1,436 @@
+        literal("txt"),
+        literal("html")
+      )
+
+      let :cli_config, object(
+        baseline: string?,
+        config: array?(string),
+        "config-resource": array?(string),
+        "disable-default-rulesets": boolean?,
+        excludes: array?(string),
+        includes: array?(string),
+        input: array?(string),
+        "language-version": number?
+      )
+
+      let :gradle_config, object(
+        task: string,
+        report_id: report_ids,
+        report_path: string
+      )
+
+      let :maven_config, object(
+        phase: string,
+        report_id: report_ids,
+        report_path: string
+      )
+
+      let :cli, Schema::RunnerConfig.base.update_fields {|hash| hash[:cli] = cli_config}
+      let :gradle, Schema::RunnerConfig.base.update_fields {|hash| hash[:gradle] = gradle_config}
+      let :maven, Schema::RunnerConfig.base.update_fields {|hash| hash[:maven] = maven_config }
+      let :base, Schema::RunnerConfig.base
+
+      let :runner_config, enum(
+        cli, gradle, maven, base,
+        detector: -> (value) {
+          case
+          when value.key?(:cli)
+            cli
+          when value.key?(:gradle)
+            gradle
+          when value.key?(:maven)
+            maven
+          else
+            base
+          end
+        }
+      )
+    end
+
+    def self.ci_config_section_name
+      'detekt'
+    end
+
+    def analyzer_name
+      'detekt'
+    end
+
+    def analyze(changes)
+      ensure_runner_config_schema(Schema.runner_config) do |config|
+        delete_unchanged_files changes, only: [".kt", ".kts"]
+
+        check_runner_config(config) do |checked_config|
+          @detekt_config = checked_config
+
+          issues = case
+                   when gradle_config
+                     run_gradle
+                   when maven_config
+                     run_maven
+                   else
+                     run_cli
+                   end
+
+          Results::Success.new(guid: guid, analyzer: analyzer).tap do |result|
+            issues.each do |issue|
+              result.add_issue issue
+            end
+          end
+        end
+      end
+    end
+
+    def analyzer_version
+      unknown_version = "0.0.0"
+
+      @analyzer_version ||=
+        case
+        when gradle_config
+          extract_detekt_version_for_gradle || unknown_version
+        when maven_config
+          extract_detekt_version_for_maven || unknown_version
+        else
+          pom_dir = Pathname(ENV['RUNNER_USER_HOME']) / analyzer_name
+          extract_detekt_version_for_maven(pom_dir) || unknown_version
+        end
+    end
+
+    def detekt_config
+      @detekt_config
+    end
+
+    def cli_config
+      detekt_config[:cli]
+    end
+
+    def gradle_config
+      detekt_config[:gradle]
+    end
+
+    def maven_config
+      detekt_config[:maven]
+    end
+
+    def run_cli
+      report_id = "xml"
+      report_path = "detekt-report.xml"
```

Comment:
[suggestion] What about using [`Tempfile.new`](https://ruby-doc.org/stdlib-2.7.0/libdoc/tempfile/rdoc/Tempfile.html#method-c-new)? This does not affect on the current directory.

```suggestion
      report_path = Tempfile.new("detekt-report")
```